domino/lib/src/experimental/idom_server.dart

```diff
@@ -0,0 +1,250 @@
+import 'dart:convert';
+
+import 'idom.dart';
+
+final _attrEscaper = HtmlEscape(HtmlEscapeMode.attribute);
+final _textEscaper = HtmlEscape(HtmlEscapeMode.element);
+
+class ServerDomContext implements DomContext<_IdomElem, Function> {
+  final _IdomElem _rootElem;
+  final _path = <_IdomElem>[];
+  final _shdPath = <_IdomElem>[];
+  final _index = <int>[];
+
+  @override
+  _IdomElem get element => _path.last;
+  _IdomElem get _shdElement => _shdPath.last;
+  @override
+  _IdomNode get pointer => _index.last < _shdElement.nodes.length
+      ? _shdElement.nodes[_index.last]
+      : null;
+
+  ServerDomContext([_IdomElem root]) : _rootElem = root ?? _IdomElem(null) {
+    _index.add(0);
+    _path.add(_rootElem);
+    _shdPath.add(_rootElem);
+  }
+
+  // TODO: onCreate, onRemove?
```

Comment:
`onCreate` and `onRemove` can be ignored here, they are for dynamic browser context.